sql/core/src/test/scala/org/apache/spark/status/api/v1/sql/SqlCompileResourceSuite.scala

```diff
@@ -0,0 +1,119 @@
+      val count = getDF().count()
+      assert(count == 2, s"Expected Query Count is 2 but received: $count")
+    }
+
+    // Spark apps launched by local-mode seems not having `attemptId` as default
+    // so UT is just added for existing endpoints.
+    val executionId = callCompilerStatRestEndpointAndVerifyResult()
+    callCompilerStatRestEndpointByExecutionIdAndVerifyResult(executionId)
+  }
+
+  private def callCompilerStatRestEndpointAndVerifyResult(): Long = {
+    val url = new URL(spark.sparkContext.ui.get.webUrl
+      + s"/api/v1/applications/${spark.sparkContext.applicationId}/compiler")
+    val jsonResult = verifyAndGetCompileStatRestResult(url)
+    val compilerStats = JsonMethods.parse(jsonResult).extract[Seq[CompileData]]
+    assert(compilerStats.size > 0,
+      s"Expected Query Result Size is higher than 0 but received: ${compilerStats.size}")
+    val compilerStatData = compilerStats.head
+    verifyCompilerStatRestContent(compilerStatData)
+    compilerStatData.id
+  }
+
+  private def callCompilerStatRestEndpointByExecutionIdAndVerifyResult(executionId: Long): Unit = {
+    val url = new URL(spark.sparkContext.ui.get.webUrl
+      + s"/api/v1/applications/${spark.sparkContext.applicationId}/compiler/${executionId}")
+    val jsonResult = verifyAndGetCompileStatRestResult(url)
+    val compilerStatData = JsonMethods.parse(jsonResult).extract[CompileData]
+    verifyCompilerStatRestContent(compilerStatData)
+  }
+
+  private def verifyCompilerStatRestContent(compileStats: CompileData): Unit = {
+    assert(compileStats.appID != "", "Should have valid application ID")
+    assert(compileStats.id > -1,
+      s"Expected execution ID is valid, found ${compileStats.id}")
+    assert(compileStats.ruleInfo.length == 50,
+      s"Expected number of Spark rules is 50, found ${compileStats.ruleInfo.length}")
+    assert(compileStats.phaseInfo.length == 3,
+      s"Expected number of phase info is: 3 found ${compileStats.phaseInfo.length}")
+  }
+
+  private def verifyAndGetCompileStatRestResult(url: URL): String = {
+    val (code, resultOpt, error) = getContentAndCode(url)
+    assert(code == 200, s"Expected Http Response Code is 200 but received: $code for url: $url")
+    assert(resultOpt.nonEmpty, s"Rest result should not be empty for url: $url")
+    assert(error.isEmpty, s"Error message should be empty for url: $url")
+    resultOpt.get
+  }
+
+  private def getDF(): DataFrame = {
+    val person: DataFrame =
+      spark.sparkContext.parallelize(
+        Person(0, "mike", 30) ::
+          Person(1, "jim", 20) :: Nil).toDF()
+
+    val salary: DataFrame =
+      spark.sparkContext.parallelize(
+        Salary(0, 2000.0) ::
+          Salary(1, 1000.0) :: Nil).toDF()
+
+    val salaryDF = salary.withColumnRenamed("personId", "id")
+    val ds = person.join(salaryDF, "id")
+      .groupBy("name", "age", "salary").avg("age", "salary")
+      .filter(_.getAs[Int]("age") <= 30)
```

Comment:
the filtering filters nothing with the provided two persons. Is this intentional ?

Reply:
Yes